File: Decorator Pattern/PHP/Decorator.php
<?php

require_once "Base.php";
require_once "Obj.php";

abstract class Decorator implements Base 
{
    protected int $id;
    protected Obj $newObj;

    public function getID()
    {
        return $this->id;
    }

    public abstract function getInfo();
}


?>

File: Decorator Pattern/CS/Decorator.cs
namespace DecoratorPattern
{
    public abstract class Decorator : NormalCar
    {
        private readonly bool _registered;
        
        protected Decorator(string model, string number, bool registered) : base(model, number)
        {
            _registered = registered;
        }

        public bool IsRegistered()
        {
            return this._registered;
        }
    }
}